cosi.proto

```diff
@@ -74,6 +74,59 @@ service Provisioner {
     rpc ProvisionerRevokeBucketAccess (ProvisionerRevokeBucketAccessRequest) returns (ProvisionerRevokeBucketAccessResponse);
 }
 
+enum ProtocolName {
+    UnknownProtocol = 0;
+    S3 = 1;
+    AZURE = 2;
+    GCS = 3;
+}
+
+enum S3SignatureVersion {
+    UnknownSignature = 0;
+    S3V2 = 1;
+    S3V4 = 2;
+}
+
+enum AnonymousBucketAccessMode {
+    UnknownBucketAccessMode = 0;
+    // Default, disallow uncredentialed access to the backend storage.
+    Private = 1;
+    // Read only, uncredentialed users can call ListBucket and GetObject.
+    ReadOnly = 2;
+    // Write only, uncredentialed users can only call PutObject.
+    WriteOnly = 3;
+    // Read/Write, same as ro with the addition of PutObject being allowed.
+    ReadWrite = 4;
+}
+
+message S3Parameters {
+    string endpoint = 1;
+    string bucket_name = 2;
+    string region = 3;
+    S3SignatureVersion signature_version = 4;
```

Comment:
What is `S3SignatureVersion` how is it different from `string version` below?
Please add comments for everything.